### DISCUSSION TITLE
Appointments, Todos, and Tasks

### DISCUSSION THREAD

**Discussion by mikeberger** [mikeberger/borg_calendar#142] [General]
So, what are appointments, todos, and tasks in BORG and what do they correspond to in other apps?

My earliest BORG appointment was created on 11/28/1990 when BORG was a C++ program. BORG came before Caldav. It came before Yahoo Calendar (or even Yahoo). It came before Windows 3.1. Therefore the objects in BORG and their data fields don't exactly match up with the events, todos, and tasks in outlook, google calendar, caldav, etc.... BORG predates them all.

FYI - Caldav is a calendaring standard that various calendar programs sync to in order to share calendars. BORG is one of many calendars that sync with it. 

Events (aka Appointments) in BORG correspond to Events in other calendars and they share most, but not all of the data fields and capabilities found in Caldav events. BORG can sync events/appointments with Caldav and can coexist pretty well with other caldav calendar clients. Events are pretty straightforward. In caldav, they are VEVENTS. One problem with Caldav in general is that calendar clients, including BORG, do not implement the same set of Event features. Therefore, using features that are only found in a particular calendar program can cause problems when syncing across calendar programs.

Todos in BORG correspond to VTODOs in Caldav **_and are referred to as TASKS or REMINDERS in most other apps_**. BORG appointments that are marked as todos can sync with android tasks, given the right sync software (http://dmfs.org/caldav/) which can convert VTODOs in caldav to a something that certain apps can understand as tasks. Some programs, like Mozilla Lightning, treat VTODOs very well. On the other hand, google calendar does not handle them. On Android, if CalDAV-Sync is used to sync with Caldav, then VTODOs can be seen in apps such as aCalendar+ or Open Tasks.

**_Projects and Tasks in BORG correspond to nothing in Caldav_**. BORG can create "dummy" items in caldav for Projects, Tasks, and Subtasks that have a due date so that something shows up in the caldav task (VTODO)  list - but these are just dummy records that are missing most of the BORG data. BORG Projects and Tasks best correspond to items in an issue tracking system such as Jira. BORG Projects and Tasks were originally used to track software development work in the 1990's, before they were generalized.

TLDR

BORG Appointments = Calendar Events
BORG Todos = Tasks/reminders  in other apps
BORG Projects and Tasks = Jira Issues
BORG syncs Appointments and Todos with Caldav.

**Comment by ghost** [comment]
BORG Projects and Tasks should optionally sync with either  Github  Gitlab or others .. Gitea ......... 
or my favorite is "kanboard"
https://github.com/kanboard/kanboard/  :-)
